client/src/components/Attachment/UploadAttachment.js

```diff
@@ -0,0 +1,136 @@
+  const [error, setError] = useState(null)
+  const [remove, setRemove] = useState(false)
+  const [uploadedList, setUploadedList] = useState([])
+  const relatedObject = getRelatedObject()
+
+  const handleUploadFile = file => {
+    setUploadedList(current => [...current, file])
+    toast.success("Your document has been uploaded")
+  }
+
+  const AttachmentSave = async e => {
+    const file = e.target.files[0]
+    const selectedAttachment = new Attachment({
+      fileName: file.name,
+      mimeType: file.type,
+      contentLength: file.size,
+      attachmentRelatedObjects: [
+        {
+          relatedObjectType: relatedObject.type,
+          relatedObjectUuid: relatedObject.uuid
+        }
+      ],
+      classification:
+        Settings.fields.attachment.classification.choices.UNDEFINED.value
+    })
+
+    return save(selectedAttachment, relatedObject.uuid, false)
+      .then(response => {
+        selectedAttachment.uuid = response.createAttachment
+        return axios
+          .postForm(
+            `/api/attachment/uploadAttachmentContent/${selectedAttachment.uuid}`,
+            { file }
+          )
+          .then(() => handleUploadFile(selectedAttachment))
+          .catch(() => toast.error("Attachment content upload failed"))
+      })
+      .catch(error => toast.error(`Attachment upload failed: ${error.message}`))
+  }
+
+  const handleFileEvent = async e => {
+    // Control related object has an uuid or not
+    if (!relatedObject.uuid) {
+      saveAttachment().then(async response => {
+        relatedObject.uuid = response
+        await AttachmentSave(e)
+      })
+    } else {
+      await AttachmentSave(e)
+    }
+  }
+
+  return (
+    <div>
+      <Messages error={error} />
+      {/** **** Select and drop file in here **** **/}
+      <section className="FileUploadContainer">
+        <div>
+          <p className="DragDropText">
+            <Icon className="uploadIcon" size={30} icon={IconNames.EXPORT} />
+            <span style={{ marginTop: "5px" }}>
+              <b>Choose a file</b> or Drag it here
+            </span>
+          </p>
+        </div>
+        <input
+          className="FormField"
+          id="fileUpload"
+          type="file"
+          accept="image/*, .pdf"
+          onChange={handleFileEvent}
+        />
+      </section>
+
+      {/** **** Show uploaded files in here **** **/}
+      <div style={{ display: "flex", flexWrap: "wrap" }}>
+        {uploadedList.map((attachment, index) => (
+          <AttachmentCard
+            key={index}
+            attachment={attachment}
+            index={index}
+            remove={remove}
+            setError={setError}
+            setRemove={setRemove}
+            uploadedList={uploadedList}
+            setUploadedList={setUploadedList}
+          />
+        ))}
+        {/** For edit report edit page show uploaded attachments **/}
+        {edit && <UploadedAttachments uuid={relatedObject.uuid} />}
```

Comment:
Actually, it is not really out of place because when we place it on the form page, extra changes will be required and it is more compatible with the uploadAttachment component as a structure, so I think it's better to have it there.

Reply:
Done.